service/storage.go

```diff
@@ -43,13 +43,17 @@ func (gs *GCSStorage) Store(data []byte, filename string) (string, error) {
 	writer := obj.NewWriter(ctx)
 	writer.ContentType = "text/plain"
 	if _, err := writer.Write(data); err != nil {
-		gs.logger.WithError(err).Info("there was some error writing file")
+		gs.logger.WithError(err).Error("there was some error writing file")
 		return "", err
 	}
 	if err := writer.Close(); err != nil {
-		gs.logger.WithError(err).Info("error closing writer ")
+		gs.logger.WithError(err).Error("error closing writer ")
+		return "", err
+	}
+	if err := obj.ACL().Set(ctx, storage.AllUsers, storage.RoleReader); err != nil {
```

Comment:
You can define the ACL in the writer and make one less operation in the GCS API :D 

```
writer.ObjectAttrs.ACL = []storage.ACLRule{{Entity: storage.AllUsers, Role: storage.RoleReader}}
```